modules/jwplayerBidAdapter.js

```diff
@@ -50,4 +90,110 @@ export const spec = {
   // onBidderError: function({ error, bidderRequest }) {}
 };
 
+function buildRequest(bidRequest, bidderRequest) {
+  const openrtbRequest = {
+    id: bidRequest.bidId,
+    imp: buildRequestImpression(bidRequest, bidderRequest),
+    site: buildRequestSite(bidRequest),
+    device: buildRequestDevice()
+  };
+
+  // Attaching GDPR Consent Params
+  if (bidderRequest.gdprConsent) {
+    deepSetValue(openrtbRequest, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    deepSetValue(openrtbRequest, 'regs.ext.gdpr', (bidderRequest.gdprConsent.gdprApplies ? 1 : 0));
+  }
+
+  // CCPA
+  if (bidderRequest.uspConsent) {
+    deepSetValue(openrtbRequest, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+  }
+
+  return JSON.stringify(openrtbRequest);
+}
+
+function buildRequestImpression(bidRequest) {
+  const impressionObject = {
+    id: bidRequest.adUnitCode,
+  };
+
+  impressionObject.video = buildImpressionVideo(bidRequest);
+
+  const bidFloorData = buildBidFloorData(bidRequest);
+  if (bidFloorData) {
+    impressionObject.bidfloor = bidFloorData.floor;
+    impressionObject.bidfloorcur = bidFloorData.currency;
+  }
+
+  impressionObject.ext = buildImpressionExtension(bidRequest);
+
+  return [impressionObject];
+}
+
+function buildImpressionVideo(bidRequest) {
+  const videoParams = deepAccess(bidRequest, 'mediaTypes.video', {});
+
+  const video = {};
+
+  VIDEO_ORTB_PARAMS.forEach((param) => {
+    if (videoParams.hasOwnProperty(param)) {
+      video[param] = videoParams[param];
+    }
+  });
+
+  return video;
+}
+
+function buildImpressionExtension(bidRequest) {
+  return {
+    appnexus: {
+      placement_id: bidRequest.params.placementId
+    }
+  };
+}
+
+function buildBidFloorData(bidRequest) {
+  const {params} = bidRequest;
+  const currency = params.currency || 'USD';
+
+  let floorData;
+  if (isFn(bidRequest.getFloor)) {
+    const bidFloorRequest = {
+      currency: currency,
+      mediaType: VIDEO,
+      size: '*'
+    };
+    floorData = bidRequest.getFloor(bidFloorRequest);
+  } else if (params.bidfloor) {
+    floorData = {floor: params.bidfloor, currency: currency};
+  }
+
+  return floorData;
+}
+
+function buildRequestSite(bidRequest) {
+  const configSite = config.getConfig('ortb2.site');
+
+  const site = {
+    domain: (configSite && configSite.domain) ? configSite.domain : window.location.hostname,
+    page: (configSite && configSite.page) ? configSite.page : window.location.href,
+    ref: bidRequest.refererInfo ? bidRequest.refererInfo.referer || null : null
+  };
+
+  // Site Content
+  if (configSite && configSite.content && isPlainObject(configSite.content)) {
+    site.content = {};
+    for (const contentKey in configSite.content) {
+      site.content[contentKey] = configSite.content[contentKey];
```

Comment:
why the loop instead of an assignment ? Seems like you're looping through all of the properties, which is the same as assigning. Assignment is better because no loops and less code

Reply:
done.